contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -118,6 +141,50 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
     emit WatchlistUpdated();
   }
 
+  /// @notice Adds a new address to the watchlist
+  /// @param targetAddress the address to be added to the watchlist
+  /// @param dstChainSelector carries a non-zero value in case the targetAddress is an onRamp, otherwise it carries a 0
+  /// @dev this function has to be compatible with the event onRampSet(address, dstChainSelector) emitted by
+  /// the CCIP router. Important detail to know is this event is also emitted when an onRamp is decomissioned,
+  /// in which case it will carry the proper dstChainSelector along with the 0x0 address
+  function addToWatchList(address targetAddress, uint64 dstChainSelector) public onlyRoleOrAdminRole(EXECUTOR_ROLE) {
+    if (s_targets[targetAddress].isActive) revert DuplicateAddress(targetAddress);
+    address oldAddress = s_onRampAddresses[dstChainSelector];
+    // if targetAddress is an existing onRamp, there's a need of cleaning the previous onRamp associated to this dstChainSelector
+    // there's no need to remove any other address that's not an onRamp
+    if (dstChainSelector > 0 && bytes(abi.encodePacked(oldAddress)).length > 0) {
+      removeFromWatchList(oldAddress);
+    }
+    // only add the new address if it's not 0x0
+    if (targetAddress != address(0)) {
+      s_onRampAddresses[dstChainSelector] = targetAddress;
+      s_targets[targetAddress] = MonitoredAddress({
+        isActive: true,
+        minBalance: DEFAULT_MIN_BALANCE,
+        topUpAmount: DEFAULT_TOP_UP_AMOUNT,
+        lastTopUpTimestamp: 0
+      });
+      s_watchList.push(targetAddress);
+    } else {
+      // if the address is 0x0, it means the onRamp has ben decomissioned and has to be cleaned
+      delete s_onRampAddresses[dstChainSelector];
```

Comment:
this feels like a non-obvious use case for the addToWatchList() function. Would it make more sense to move this functionality to another public function like `decomissionOnRamp(address target)` or something similar? Then in `addToWatchlist(..)` you could `require(targetAddess != address(0))`.

Reply:
The problem behind this, which I agree with you that this is totally non-obvious, is that this function has to be triggered by a CL Automation upkeep, and while doing so preserving the sanity of the watchlist.
CL Automation would trigger in onRampSet(dstChainSelector, address) events, and the gotcha here is when an onRamp is decomissioned the onRampSet carries the dstChainSelector and address(0), meaning we have to remove the onRamp completely. This is the only corner case this would be triggered.

Reply:
ah I see! Maybe just a different function name would suffice then? Something that more accurately describes the intended use case.